### PR TITLE
Check for possible relative paths in system()

### PR DESCRIPTION
**Pull request by riataman**
Implements this: https://github.com/riboseinc/retrace/issues/163

I figured that even when it may not catch 100% of the cases (since system() basically dumps the command into the shell and that has a lot of escaping, expansions and such) we might as well implement the most common ones and word the warning as "MAY contain a relative path".

### CODE REVIEW

**Comment by ronaldtse** [comment]
@erikbor your comments please 👍

File: src/exec.c [hunk 1]
```diff
@@ -112,6 +113,59 @@ inject_retrace_env_vars(const char **env)
...
+				}
+
+				s = real_strchr(s + 1, '|');
+			}
+		}
+	}
+
+	return retval;
+}
+
```
Comment (drystone) [inline src/exec.c]:
```
static int
may_contain_relative_paths(const char *s)
{
    while (s != NULL) {
        while (*s == ';' || *s == '|' || isspace(*s))
            ++s;
        if (*s && *s != '/')
            return 1; /* could check against list of builtins here ? */
        s = real_strpbrk(s, ";|");
    }
    return 0;
}
```

**Comment by riataman** [comment]
I would take a guess and think that the majority of the cases are a single command and that's it. So the simplest check is checking if the command starts with a / or not.

The next more common case I guess are multiple commands (check a / after a semicolon) and pipes (check for / after |)

Again taking a wild guess I'm thinking that should covert the vast majority  ofl the cases you might see in the wild AND the code seems very simple to me.

Attempting to cover 100% of the cases would be pretty much impossible, since it depends on what shell the user is running and that can be anything.

**Comment by erikbor** [comment]
@riataman nice one Pablo!

**Pull request merged**